src/shims/env.rs

```diff
@@ -111,4 +112,31 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             Ok(-1)
         }
     }
+
+    fn getcwd(
+        &mut self,
+        buf_op: OpTy<'tcx, Tag>,
+        size_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Scalar<Tag>> {
+        let this = self.eval_context_mut();
+
+        let buf = this.read_scalar(buf_op)?.not_undef()?.assert_ptr();
+        let size = this.read_scalar(size_op)?.to_usize(&*this.tcx)?;
+        // If we cannot get the current directory, we return null
+        if let Ok(cwd) = env::current_dir() {
+            // It is not clear what happens with non-utf8 paths here
+            let mut bytes = cwd.display().to_string().into_bytes();
+            // If the buffer is smaller than the path, we return null
+            if bytes.len() as u64 <= size {
+                bytes.resize(size as usize, 0);
+                // Temporary pointer to write the path into miri's memory
+                let ptr = this.memory_mut().allocate_static_bytes(bytes.as_slice(), MiriMemoryKind::Env.into());
+                this.memory_mut().copy(ptr, buf, Size::from_bytes(size), true)?;
```

Comment:
You should be able to get_mut the memory and write directly to the slice. There are methods on Memory to do that just for a specific pointer and number of bytes